### PR TITLE
Add support for named service tokens for logger metadata

### PR DESCRIPTION
**Pull request by brettlyman**
So our logging can display which service connected/authenticated

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -21,6 +21,10 @@ use Mix.Config
 #     config :logger, level: :info
 #
 
+config :logger, :console,
+  format: "\n$time $metadata [$level] $message",
```
Comment (sethgw) [inline config/config.exs:25]: Will this conflict with the config in BM?  Will BM's format overwrite this?

File: config/config.exs [hunk 2]
```diff
@@ -21,6 +21,10 @@ use Mix.Config
...
+  metadata: [:service_name]
```
Comment (sethgw) [inline config/config.exs:26]: Same question about BM's config.  Will the config merge or is it up to BM to add `[:service_name]`

Comment (brettlyman) [reply config/config.exs:26]: I'll need to research how "third party configs" work in Elixir. Great concern....I'll do some tests and get back to ya...

Comment (brettlyman) [reply config/config.exs:26]: image

Comment (brettlyman) [reply config/config.exs:26]: I.e. I don't think the config file from the dependency matters unless you actually run `iex` within that module.

File: config/config.exs (hunk 1 above)
Comment (brettlyman) [reply config/config.exs:25]: See response below -- the config from a dependency doesn't get used/imported, only the config of the actual project you're compiling/running.

File: config/config.exs (hunk 2 above)
Comment (brettlyman) [reply config/config.exs:26]: To answer the second question: BM will need to add `[:service_name]` to the metadata config.

Comment (sethgw) [reply config/config.exs:26]: 🙇

File: lib/simple_token_authentication.ex [hunk 3]
```diff
@@ -36,6 +42,15 @@ defmodule SimpleTokenAuthentication do
 
   defp matches?(token, value) when is_binary(token) and is_binary(value),
```
Comment (sethgw) [inline lib/simple_token_authentication.ex:43]: I like that you're keeping backwards compatibility!

**Pull request merged**